core/src/main/java/hudson/model/Queue.java

```diff
@@ -1630,6 +1632,10 @@ public void maintain() {
                     continue;
                 }
 
+                Label itemLabel = p.getAssignedLabel();
+                if (noCandidateLabels.contains(itemLabel)) {
+                    continue;
```

Comment:
Doing this changes some behaviour. 

Example:
```
p.transientCausesOfBlockage
```

will no longer get assigned on all buildable items.